config/config-zk/src/main/java/com/alipay/sofa/rpc/dynamic/zk/ZookeeperDynamicConfigManager.java

```diff
@@ -0,0 +1,112 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alipay.sofa.rpc.dynamic.zk;
+
+import com.alipay.sofa.rpc.auth.AuthRuleGroup;
+import com.alipay.sofa.rpc.common.utils.StringUtils;
+import com.alipay.sofa.rpc.dynamic.DynamicConfigKeyHelper;
+import com.alipay.sofa.rpc.dynamic.DynamicConfigManager;
+import com.alipay.sofa.rpc.dynamic.DynamicHelper;
+import com.alipay.sofa.rpc.ext.Extension;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+
+/**
+ * @author Narziss
+ * @version ZookeeperDynamicConfigManager.java, v 0.1 2024年07月20日 09:23 Narziss
+ */
+
+@Extension(value = "zookeeper", override = true)
+public class ZookeeperDynamicConfigManager extends DynamicConfigManager {
+
+    private CuratorFramework zkClient;
+    private static final String ADDRESS = "127.0.0.1:2181";
+    private static final String DEFAULT_NAMESPACE = "sofa-rpc";
+    private static final String CONFIG_NODE = "/config";
+    private static final String Default_GROUP = "/application";
+
+    protected ZookeeperDynamicConfigManager(String appName) {
+        super(appName);
+        zkClient = CuratorFrameworkFactory.builder()
+                .connectString(ADDRESS)
+                .retryPolicy(new ExponentialBackoffRetry(1000, 3))
+                .namespace(DEFAULT_NAMESPACE)
+                .build();
+        zkClient.start();
+    }
+
+    @Override
+    public void initServiceConfiguration(String service) {
+        //TODO not now
+    }
+
+    @Override
+    public String getProviderServiceProperty(String service, String key) {
+        try {
+            String path = getPrefixPath() + DynamicConfigKeyHelper.buildProviderServiceProKey(service, key);
+            byte[] bytes = zkClient.getData().forPath(path);
+            return bytes != null ? new String(bytes) : DynamicHelper.DEFAULT_DYNAMIC_VALUE;
+        } catch (Exception e) {
+            return DynamicHelper.DEFAULT_DYNAMIC_VALUE;
+        }
+    }
+
+    @Override
+    public String getConsumerServiceProperty(String service, String key) {
+        try {
+            String path = getPrefixPath() + DynamicConfigKeyHelper.buildConsumerServiceProKey(service, key);
```

Comment:
This `zkClient.getData().forPath`  method does not have cached. Here should add a cache in conjunction with the listener mentioned above.